model/convert/service_event.go

```diff
@@ -326,75 +487,150 @@ func convertParticipants(cdcParticipants []cadence.Value) (flow.IdentityList, er
 				cadence.Struct{},
 			)
 		}
-		cdcNodeInfoFields := cdcNodeInfoStruct.Fields
 
-		expectedFields := 14
-		if len(cdcNodeInfoFields) < expectedFields {
+		expectedFieldCount := 14
+		if len(cdcNodeInfoStruct.Fields) < expectedFieldCount {
 			return nil, fmt.Errorf(
 				"insufficient fields (%d < %d)",
-				len(cdcNodeInfoFields),
-				expectedFields,
+				len(cdcNodeInfoStruct.Fields),
+				expectedFieldCount,
 			)
 		}
 
-		// create and assign fields to identity from cadence Struct
-		identity := new(flow.Identity)
-		role, ok := cdcNodeInfoFields[1].(cadence.UInt8)
-		if !ok {
-			return nil, invalidCadenceTypeError(
-				"nodeInfo.role",
-				cdcNodeInfoFields[1],
-				cadence.UInt8(0),
+		if cdcNodeInfoStruct.Type() == nil {
+			return nil, fmt.Errorf("nodeInfo struct doesn't have type")
+		}
+
+		cdcNodeInfoStructType := cdcNodeInfoStruct.Type().(*cadence.StructType)
+
+		var nodeIDHex cadence.String
+		var role cadence.UInt8
+		var address cadence.String
+		var networkKeyHex cadence.String
+		var stakingKeyHex cadence.String
+		var initialWeight cadence.UInt64
+		var foundFieldCount int
+
+		for i, f := range cdcNodeInfoStructType.Fields {
+			switch f.Identifier {
+			case "id":
+				foundFieldCount++
+				nodeIDHex, ok = cdcNodeInfoStruct.Fields[i].(cadence.String)
+				if !ok {
+					return nil, invalidCadenceTypeError(
+						"nodeInfo.id",
+						cdcNodeInfoStruct.Fields[i],
+						cadence.String(""),
+					)
+				}
+
+			case "role":
+				foundFieldCount++
+				role, ok = cdcNodeInfoStruct.Fields[i].(cadence.UInt8)
+				if !ok {
+					return nil, invalidCadenceTypeError(
+						"nodeInfo.role",
+						cdcNodeInfoStruct.Fields[i],
+						cadence.UInt8(0),
+					)
+				}
+
+			case "networkingAddress":
+				foundFieldCount++
+				address, ok = cdcNodeInfoStruct.Fields[i].(cadence.String)
+				if !ok {
+					return nil, invalidCadenceTypeError(
+						"nodeInfo.networkingAddress",
+						cdcNodeInfoStruct.Fields[i],
+						cadence.String(""),
+					)
+				}
+
+			case "networkingKey":
+				foundFieldCount++
+				networkKeyHex, ok = cdcNodeInfoStruct.Fields[i].(cadence.String)
+				if !ok {
+					return nil, invalidCadenceTypeError(
+						"nodeInfo.networkingKey",
+						cdcNodeInfoStruct.Fields[i],
+						cadence.String(""),
+					)
+				}
+
+			case "stakingKey":
+				foundFieldCount++
+				stakingKeyHex, ok = cdcNodeInfoStruct.Fields[i].(cadence.String)
+				if !ok {
+					return nil, invalidCadenceTypeError(
+						"nodeInfo.stakingKey",
+						cdcNodeInfoStruct.Fields[i],
+						cadence.String(""),
+					)
+				}
+
+			case "tokensStaked":
```

Comment:
don't count these fields since you're ignoring them.  change the expectedFieldCount instead (same elsewhere)